AmplifyPlugins/API/Sources/AWSAPIPlugin/Interceptor/RequestInterceptor/APIKeyURLRequestInterceptor.swift

```diff
@@ -11,19 +11,19 @@ import Foundation
 
 struct APIKeyURLRequestInterceptor: URLRequestInterceptor {
 
+    let userAgent: String
     let apiKeyProvider: APIKeyProvider
 
     init(apiKeyProvider: APIKeyProvider) {
         self.apiKeyProvider = apiKeyProvider
+        self.userAgent = AmplifyAWSServiceConfiguration.frameworkMetaData().description
```

Comment:
nit: Why not just have these user agents initialized on declaration (and `private`)?

```swift
private let userAgent = AmplifyAWSServiceConfiguration.frameworkMetaData().description
```

This is totally just a pet peeve of mine :P

Reply:
Alternatively inject it w/ a default argument to make testing easier.
```swift
init(
    apiKeyProvider: APIKeyProvider,
    userAgent: String = AmplifyAWSServiceConfiguration.frameworkMetaData().description
)
```

Reply:
opted to update per @ruisebas suggestion and made private variable that initializes the value at declaration.